### DISCUSSION TITLE
MSHV on Windows

### DISCUSSION THREAD

**Discussion by yavulan** [cloud-hypervisor/cloud-hypervisor#3453] [Q&A]
Docs states that "cloud-hypervisor ... runs on top of KVM and the MSHV hypervisors ."

How to use cloud-hypervisor on Windows to run on top of MSHV hypervisor (windows target for cargo build seems to be unsupported)?

Thank you.

**Comment by rbradford** [comment]
@liuw Do you have a reference to a document explaining what MSHV is?

**Comment by liuw** [comment]
MSHV is short for "Microsoft Hypervisor". It is a standalone binary. Conceptually it is not tied to Windows.

A virtualization stack consists of a hypervisor, a host kernel and user space components running on the controlling domain. We only ported Cloud Hypervisor to run on Linux, not Windows.

It should be possible to port Cloud Hypervisor to run on Windows by using the Windows Hypervisor Platform APIs (https://docs.microsoft.com/en-us/virtualization/api/hypervisor-platform/hypervisor-platform). In that case the underlying hypervisor is still MSHV, but the host kernel and user space components come from Windows. It is something we have on our radar, but we're not going to work on that any time soon. In any case, if someone wants to step up and do that work, we can talk how it can be done.

**Reply by ecki** [reply]
So it is enough to start Linux on a Hyper-V host and use c-h to use it instead of kvm, or does it need special host kernel setup? Does it work with Hyper-v in Windows Server?

**Reply by liuw** [reply]
It still needs some yet to be open-sourced components (kernel and loader). We're working on it.
